internal/entities/currency.go

```diff
@@ -1,27 +1,30 @@
 package entities
 
+import "strings"
+
 type Currency string
 
 const (
 	AmericanDollar   Currency = "USD"
 	UkrainianHryvnia Currency = "UAH"
 )
 
-var SupportedCurrencies = []Currency{AmericanDollar, UkrainianHryvnia}
+var SupportedCurrencies = map[Currency]struct{}{
+	AmericanDollar:   {},
+	UkrainianHryvnia: {},
+}
 
 func (c Currency) IsSupported() bool {
-	for _, cur := range SupportedCurrencies {
-		if c == cur {
-			return true
-		}
-	}
+	_, ok := SupportedCurrencies[c]
 
-	return false
+	return ok
 }
-func FromString(data []string) []Currency {
+
+// CurrenciesFromString converts []string to []Currency.
+func CurrenciesFromString(data []string) []Currency {
```

Comment:
Does it make sense do be in domain?
Here you doesn't make any validation or some logic, so I guess here it's redundant.

Reply:
Replaced `CurrenciesFromString([]string) []Currency` with `ValidateCurrencies([]string) error`